### PR TITLE
Lint: nullPointerRedundantCheck

### PR DESCRIPTION
**Pull request by krader1961**
From cppcheck:

[src/lib/libast/path/pathtemp.c:217] warning (nullPointerRedundantCheck):
 Either the condition '!d' is redundant or there is possible null
 pointer dereference: d.

The issue is that the person who wrote much of this code seemed to delight
in writing clever, compact, code. So refactor it to eliminate the lint
and clarify the behavior.

### CODE REVIEW

File: src/lib/libast/path/pathtemp.c [hunk 1]
```diff
@@ -113,6 +113,18 @@ static struct Tmp_s {
     char *tmppath;
 } tmp = {.mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH};
 
+static_fn char *get_tmp_dir() {
+    char *d = astconf("TMP", NULL, NULL);
+    if (*d && eaccess(d, W_OK | X_OK) == 0) return d;
+
+    d = TMP1;
```
Comment (siteshwar) [inline src/lib/libast/path/pathtemp.c]: This assignment looks unnecessary.

File: src/lib/libast/path/pathtemp.c [hunk 2]
```diff
@@ -113,6 +113,18 @@ static struct Tmp_s {
...
+    if (eaccess(d, W_OK | X_OK) == 0) return d;
```
Comment (siteshwar) [inline src/lib/libast/path/pathtemp.c]: if (eaccess(TMP1, W_OK | X_OK) == 0) return TMP1;

File: src/lib/libast/path/pathtemp.c [hunk 3]
```diff
@@ -113,6 +113,18 @@ static struct Tmp_s {
     char *tmppath;
 } tmp = {.mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH};
 
+static_fn char *get_tmp_dir() {
```
Comment (siteshwar) [inline src/lib/libast/path/pathtemp.c]: This name might clash with other names in debug builds in future. May be rename it to `get_pathtmp_dir`.

File: src/lib/libast/path/pathtemp.c (hunk 2 above)
Comment (krader1961) [reply src/lib/libast/path/pathtemp.c]: Haha, yes. It was late and just unpacking the mess that was the original `if(...)` used up my remaining energy.

**Pull request closed without merging**